app/src/main/kotlin/com/getstrm/pace/service/DataPolicyService.kt

```diff
@@ -59,4 +64,58 @@ class DataPolicyService(
                     .setDescription("DataPolicy $id not found")
                     .build()
             )
+
+    suspend fun scanLineage(request: ScanLineageRequest): PagedCollection<LineageSummary> {
+        val policies =
+            listDataPolicies(
+                ListDataPoliciesRequest.newBuilder()
+                    .setPageParameters(request.pageParameters)
+                    .build()
+            )
+
+        // policies.map triggers this error
+        // https://stackoverflow.com/questions/57329658/getting-suspension-functions-can-be-called-only-within-coroutine-body-when-cal
+        // and I can't figure out to fix the utility function PagedCollection.map
+        val lineageSummaries =
+            policies.data.map { policy ->
+                determineLineageDataPolicies(
+                    try {
+                        processingPlatforms
+                            .getLineage(
+                                GetLineageRequest.newBuilder()
+                                    .setFqn(policy.source.ref)
+                                    .setPlatformId(policy.platform.id)
+                                    .build()
+                            )
+                            .lineageSummary
+                    } catch (e: Exception) {
+                        LineageSummary.getDefaultInstance()
```

Comment:
Is this desirable? When any issue occurs with the downstream processing platform, shouldn't we provide that as a result to the user instead of no error and just an empty lineage?

Reply:
The problem is that most platforms don't even have this implemented yet.

Reply:
I've tightened down the exception to InternalException